metrics-core/src/test/java/com/codahale/metrics/SlidingTimeWindowReservoirTest.java

```diff
@@ -24,6 +24,39 @@ public void storesMeasurementsWithDuplicateTicks() throws Exception {
     }
 
     @Test
+    public void storesMeasurementsWithDuplicateTicksTresholdNeg() throws Exception {
+        when(clock.getTick()).thenReturn(Long.MIN_VALUE / 256);
+        SlidingTimeWindowReservoir reservoir = new SlidingTimeWindowReservoir(10, TimeUnit.NANOSECONDS, clock);
+
+        reservoir.update(1);
+        reservoir.update(2);
+
+        assertThat(reservoir.getSnapshot().getValues())
+                .containsOnly(1, 2);
+    }
+
+    @Test
+    public void storesMeasurementsWithDuplicateTicksMinNeg() throws Exception {
+        when(clock.getTick()).thenReturn(Long.MIN_VALUE);
+        SlidingTimeWindowReservoir reservoir = new SlidingTimeWindowReservoir(10, TimeUnit.NANOSECONDS, clock);
+
+        reservoir.update(1);
+        reservoir.update(2);
+
+        assertThat(reservoir.getSnapshot().getValues())
+                .containsOnly(1, 2);
+    }
+
+    @Test
+    public void storesMeasurementsWithTresholdTicks() throws Exception {
```

Comment:
This test detects overflow on 417th day of server uptime (we've seen this in live and have heap dumps showing it)